atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/workflow/SingleBusyCellWorkflowsTest.java

```diff
@@ -0,0 +1,101 @@
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.factory.TransactionManagers;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.workload.store.AtlasDbTransactionStore;
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.IsolationLevel;
+import com.palantir.atlasdb.workload.store.ReadOnlyTransactionStore;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedDeleteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedReadTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedWriteTransactionAction;
+import com.palantir.atlasdb.workload.util.AtlasDbUtils;
+import com.palantir.common.concurrent.PTExecutors;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class SingleBusyCellWorkflowsTest {
+    private static final String TABLE_NAME = "busy.cell";
+    private static final int ITERATION_COUNT = 1_000;
+    private static final SingleBusyCellWorkflowConfiguration CONFIGURATION =
+            ImmutableSingleBusyCellWorkflowConfiguration.builder()
+                    .tableConfiguration(ImmutableTableConfiguration.builder()
+                            .tableName(TABLE_NAME)
+                            .isolationLevel(IsolationLevel.SERIALIZABLE)
+                            .build())
+                    .iterationCount(ITERATION_COUNT)
+                    .build();
+
+    private final InteractiveTransactionStore transactionStore = AtlasDbTransactionStore.create(
+            TransactionManagers.createInMemory(ImmutableSet.of()),
+            ImmutableMap.of(
+                    TableReference.createWithEmptyNamespace(TABLE_NAME),
+                    AtlasDbUtils.tableMetadata(IsolationLevel.SERIALIZABLE)));
+    private final Workflow workflow = SingleBusyCellWorkflows.create(
+            transactionStore,
+            CONFIGURATION,
+            MoreExecutors.listeningDecorator(PTExecutors.newSingleThreadExecutor()),
+            MoreExecutors.listeningDecorator(PTExecutors.newSingleThreadExecutor()));
+
+    @Test
+    public void workflowHistoryTransactionStoreShouldBeReadOnly() {
+        WorkflowHistory history = workflow.run();
+        assertThat(history.transactionStore()).isInstanceOf(ReadOnlyTransactionStore.class);
+    }
+
+    @Test
+    public void transactionsDoNotConflictExcessively() {
+        WorkflowHistory workflowHistory = workflow.run();
+        assertThat(workflowHistory.history()).hasSizeBetween(ITERATION_COUNT / 2, ITERATION_COUNT);
```

Comment:
Fair enough! (Though note that a bit confusingly it's actually 2 because we have both read and write single-thread executors.) I've bumped the test config to have 4 threads each for readers and writers.